### ISSUE TITLE
Current/Master GCE Discovery Docs document pre-5.1 config

### ISSUE THREAD

**Issue by darkn3rd** [elastic/elasticsearch#28908]
**Elasticsearch version** (`bin/elasticsearch --version`):
* `Version: 6.2.2, Build: 10b1edd/2018-02-16T19:01:30.685723Z, JVM: 1.8.0_161`

**Plugins installed**: (`bin/elasticsearch-plugin list`)
* `discovery-gce`

**JVM version** (`java -version`):
* `1.8.0_161`

**OS version** (`lsb_release -d` if on a Linux-like system):
* `Ubuntu 14.04.5 LTS`

**Description of the problem including expected versus actual behavior**:

The docs for Setting up GCE Discovery on both current and master, document pre-5.1 style.

* https://www.elastic.co/guide/en/elasticsearch/plugins/master/discovery-gce-usage-long.html
* https://www.elastic.co/guide/en/elasticsearch/plugins/current/discovery-gce-usage-long.html

**Actual Docs**
```yaml
cloud:
  gce:
      project_id: es-cloud
      zone: europe-west1-a
discovery:
      type: gce
```

**Expected Docs**

```yaml
cloud:
  gce:
      project_id: es-cloud
      zone: europe-west1-a
discovery:
      zen.hosts_provider: gce
```

**Comment by javanna** [comment]
cc @elastic/es-distributed

**Comment by bleskes** [comment]
@tlrx any thoughts about this?

**Comment by tlrx** [comment]
@darkn3rd Thanks for reporting this issue! It has been fixed by #28876
